src/libraries/System.Runtime/tests/System.Runtime.Tests/System/StringTests.cs

```diff
@@ -1347,36 +1347,57 @@ public static void ImplicitCast_NullString_ReturnsDefaultSpan()
+        [InlineData("Hello", 'l', 0, int.MaxValue, StringComparison.Ordinal, 2)]
+        [InlineData("Hello", 'x', 0, int.MaxValue, StringComparison.Ordinal, -1)]
+        [InlineData("Hello", 'h', 0, int.MaxValue, StringComparison.Ordinal, -1)]
+        [InlineData("Hello", 'o', 0, int.MaxValue, StringComparison.Ordinal, 4)]
+        [InlineData("Hello", 'h', 0, int.MaxValue, StringComparison.OrdinalIgnoreCase, 0)]
+        [InlineData("HelLo", 'L', 0, int.MaxValue, StringComparison.OrdinalIgnoreCase, 2)]
+        [InlineData("HelLo", 'L', 0, int.MaxValue, StringComparison.Ordinal, 3)]
+        [InlineData("HelLo", '\0', 0, int.MaxValue, StringComparison.Ordinal, -1)]
+        [InlineData("!@#$%", '%', 0, int.MaxValue, StringComparison.Ordinal, 4)]
+        [InlineData("!@#$", '!', 0, int.MaxValue, StringComparison.Ordinal, 0)]
+        [InlineData("!@#$", '@', 0, int.MaxValue, StringComparison.Ordinal, 1)]
+        [InlineData("!@#$%", '%', 0, int.MaxValue, StringComparison.OrdinalIgnoreCase, 4)]
+        [InlineData("!@#$", '!', 0, int.MaxValue, StringComparison.OrdinalIgnoreCase, 0)]
+        [InlineData("!@#$", '@', 0, int.MaxValue, StringComparison.OrdinalIgnoreCase, 1)]
+        [InlineData("_____________\u807f", '\u007f', 0, int.MaxValue, StringComparison.Ordinal, -1)]
+        [InlineData("_____________\u807f__", '\u007f', 0, int.MaxValue, StringComparison.Ordinal, -1)]
+        [InlineData("_____________\u807f\u007f_", '\u007f', 0, int.MaxValue, StringComparison.Ordinal, 14)]
+        [InlineData("__\u807f_______________", '\u007f', 0, int.MaxValue, StringComparison.Ordinal, -1)]
+        [InlineData("__\u807f___\u007f___________", '\u007f', 0, int.MaxValue, StringComparison.Ordinal, 6)]
+        [InlineData("_____________\u807f", '\u007f', 0, int.MaxValue, StringComparison.OrdinalIgnoreCase, -1)]
+        [InlineData("_____________\u807f__", '\u007f', 0, int.MaxValue, StringComparison.OrdinalIgnoreCase, -1)]
+        [InlineData("_____________\u807f\u007f_", '\u007f', 0, int.MaxValue, StringComparison.OrdinalIgnoreCase, 14)]
+        [InlineData("__\u807f_______________", '\u007f', 0, int.MaxValue, StringComparison.OrdinalIgnoreCase, -1)]
+        [InlineData("__\u807f___\u007f___________", '\u007f', 0, int.MaxValue, StringComparison.OrdinalIgnoreCase, 6)]
+        [InlineData("hello", 'e', 3, int.MaxValue, StringComparison.Ordinal, -1)]
+        [InlineData("hello", 'o', 3, int.MaxValue, StringComparison.Ordinal, 4)]
+        [InlineData("hello", 'o', 3, 0, StringComparison.Ordinal, -1)]
+        [InlineData("hello", 'o', 3, 2, StringComparison.Ordinal, 4)]
+        [InlineData("HELLO", 'o', 3, 2, StringComparison.OrdinalIgnoreCase, 4)]
+        [InlineData("abacus", 'a', 0, int.MaxValue, StringComparison.OrdinalIgnoreCase, 0)]
+        public static void IndexOf_SingleLetter_StringComparison(string s, char target, int startIndex, int count, StringComparison stringComparison, int expected)
+        {
```

Comment:
As mentioned in other comment, please add code convert the chat to a string a perform the search operation and ensure it is returning the exact result as the char case.